pkg/relayer/proxy/synchronous.go

```diff
@@ -82,18 +85,44 @@ func (sync *synchronousRPCServer) Stop(ctx context.Context) error {
 	return sync.server.Shutdown(ctx)
 }
 
-// Service returns the underlying service object.
-func (sync *synchronousRPCServer) Service() *sharedtypes.Service {
-	return sync.service
-}
-
 // ServeHTTP listens for incoming relay requests. It implements the respective
 // method of the http.Handler interface. It is called by http.ListenAndServe()
 // when synchronousRPCServer is used as an http.Handler with an http.Server.
 // (see https://pkg.go.dev/net/http#Handler)
 func (sync *synchronousRPCServer) ServeHTTP(writer http.ResponseWriter, request *http.Request) {
 	ctx := request.Context()
 
+	var originHost string
+	// Get the original host from X-Forwarded-Host header if specified in the proxy config.
+	// and fall back to the Host header if it is not specified.
+	if sync.proxyConfig.XForwardedHostLookup {
+		originHost = request.Header.Get("X-Forwarded-Host")
+	}
+
+	if originHost == "" {
+		originHost = request.Host
+	}
+
+	var supplierService *sharedtypes.Service
+	var serviceUrl *url.URL
+
+outer:
```

Comment:
I'm not a fan of using labels in production code. IMO it means we can either:
1. Use a helper that returns a couple variables.
2. Check in the outer loop if there's a value set and break then